docs/security.md

```diff
@@ -281,7 +283,58 @@ How does the client get the web service token in the first place? The client wil
+
+- Create the SSL configuration file with the following content. Typically the file is called `ssl-server.xml` but you can use any other name.
+
+```
+<configuration>
+  <property>
+    <name>ssl.server.keystore.location</name>
+    <value>/opt/apex/keystore/myapex-keystore.jks</value>
+    <description></description>
+  </property>
+  <property>
+    <name>ssl.server.keystore.keypassword</name>
+    <value>keypass2</value>
+    <description></description>
+  </property>
+  <property>
+    <name>ssl.server.keystore.password</name>
+    <value>storepass1</value>
+    <description></description>
+  </property>
+</configuration>
+```
+
+Let's assume this file will reside on each node as `/opt/apex/sslConfig/my-apex-ssl-server.xml` .
+
+- Copy the keystore file and the SSL configuration file on each Hadoop node to their designated locations and make sure they are readable by your Apex application.
+
+- Using the Apex CLI, pass the SSL configuration file location to your Apex application using the SSL_CONFIG attribute as follows:
+```
+launch -Ddt.attr.SSL_CONFIG="{\"configPath\":\"/opt/apex/sslConfig/my-apex-ssl-server.xml\"}"  <apa file>
+```
+
+- Alternatively you can use configuration files to supply the value of `dt.attr.SSL_CONFIG` attribute as described [here](http://apex.apache.org/docs/apex/application_packages/#application-packages)
+
+#### Approach 3: Using Apex CLI to Copy the Keystore
+If you cannot pre-install the SSL files on all your cluster nodes, you can use the SSL_CONFIG attribute in such a way that Apex CLI copies the keystore file from the client node to the server node and also passes on the keystore password values to the STRAM. The keystore file needs to be accessible on the Apex CLI machine so Apex CLI can copy it. Let's assume this location is `/opt/apexCli/ssl/myapex-keystore.jks`. Use the Apex CLI launch command as follows:
+```
+launch -Ddt.attr.SSL_CONFIG="{\"keyStorePath\":\"/opt/apexCli/ssl/myapex-keystore.jks\",\"keyStorePassword\":\"storepass1\",\"keyStoreKeyPassword\":\"keypass2\"}"  <apa file>
+```
+
+Apec CLI will copy the keystore file `/opt/apexCli/ssl/myapex-keystore.jks` to the destination STRAM node and also pass on the keystore password values to the STRAM. As mentioned above, you can also use configuration files to supply the value of `dt.attr.SSL_CONFIG`.
 
-We looked at the different security configuration options that are available in Apex, saw the different security requirements for distributed applications in a secure Hadoop environment in detail and looked at how the various security mechanisms in Apex solves this.
+#### Dependencies
+The use of the attribute `dt.attr.SSL_CONFIG` described in the last 2 approaches is dependent on enhancements made in YARN (https://issues.apache.org/jira/browse/YARN-6457) and Apex (https://issues.apache.org/jira/browse/APEXCORE-712) so make sure your Hadoop and Apex versions have those enhancements included.
```

Comment:
Mention of Apex enhancement included isn't really necessary - docs are versioned and this will be released with 3.7 only.